### DISCUSSION TITLE
Rename the `ISablierV2` interface

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#286] [General]
In this PR https://github.com/sablierhq/v2-core/pull/271 I suggested to rename the [ISablierV2](https://github.com/sablierhq/v2-core/blob/prb/lockup/src/interfaces/ISablierV2.sol) interface to `ISablierV2Administrative` with the idea that in the future we will have multiple products (LockUp, Payroll) and would be better to create then a common interface, `ISablierV2`, for both of them.

The options I suggest are: 
- `ISablierV2Administrative`
- `ISablierV2AdminActions`

What do you think, do you have better ideas?

**Comment by maxdesalle** [comment]
Wouldn't `ISablierV2Admin` be better in that case? It's easier to pronounce, and has the same meaning.

**Reply by PaulRBerg** [reply]
`Admin` makes one think that this is the governance contract itself, where votes are cast, and whatnot. Even if we don't end up using this name in any of our governance system contracts, I would like us not to give a false impression.

I would like the name of this contract to refer to the *modality* of being governed, and to this end I have given a few suggestions below.

**Reply by PaulRBerg** [reply]
Summoning our thesaurus master, @razgraf, into the conversation.

**Comment by PaulRBerg** [comment]
Thanks for creating this discussion, @andreivladbrg.

I agree with your proposal to change the name of the interface, though for a different reason. The `ISablierV2` name is so simplistic that it could easily mislead GitHub visitors into thinking that that is the actual protocol API. In V1, there was only `ISablier` and `Sablier`, so our early adopters may think that just by adding the `V2` suffix, they will also get the protocol functions.

It's also a mater of aesthetics - I would rather reserve the `ISablierV2` name for something more end-user-related - but as you say, Andrei, we will only be able to do this once we implement future V2 contracts.

`ISablierV2Administrative` is not bad, but I would also like to throw the following ideas around:

- `ISablierV2Administrable`
- `ISablierV2Config`
- `ISablierV2Configurable`
- `ISablierV2Governable`

**Reply by andreivladbrg** [reply]
I think the `ISablierV2Governable` is just misleading, people would think that there is a governance system behind.

I like these two: `ISablierV2Config` `ISablierV2Configurable`

**Reply by PaulRBerg** [reply]
I don't think that "Governable" is that misleading, because there is actually a governance system, it's just not decentralized at the beginning - i.e., the governance is a multisig controlled by us. However, "Governance" isn't a great choice for the reasons given [here](https://github.com/sablierhq/v2-core/discussions/286#discussioncomment-4758515).

I also like `ISablierV2Config` and `ISablierV2Configurable` better.

**Comment by razgraf** [comment]
> The ISablierV2 name is so simplistic that it could easily mislead GitHub visitors into thinking that that is the actual protocol API.

True. As I see it, this interface doesn't even define very specific "Sablier" features, but rules for our core functionality-driven contracts. I think it should have the "Sablier" particle in it, for the actual contract we're gonna deploy but additional context is necessary.

> `ISablierV2Governable`

This one is tricky, I'd leave the "Governance" part out all together, so we don't have a hard time naming things if/when we eventually introduce Governance contracts. 

---

How about something less formal like `ISablierWarden`. This are inspired from [Yield](https://github.com/yieldprotocol/vault-v2/tree/master/packages/foundry/contracts) as they seem to be using names such as `Witch` or `Cauldron`.

Does it have to abide by a certain shape e.g. "xyz-able" as in "Adminable"? What are the constraints?

**Reply by PaulRBerg** [reply]
Thanks, @razgraf. Agree with all the points I didn't reply to.

> How about something less formal like `ISablierWarden`

I am not against this style of terminology, but I don't think that "warden" is a good fit in this context.

> What are the constraints?

No hard-and-fast rules! We're just brainstorming.

**Comment by PaulRBerg** [comment]
I have gone with `SablierV2Config` and implemented the change in PR #292.

**Comment by PaulRBerg** [comment]
Locking, implemented in #292.

**Comment by PaulRBerg** [comment]
For future reference: what used to be called `SablierV2Config` in this discussion is now called `SablierV2Base`:

- https://github.com/sablier-labs/v2-core/discussions/370
- https://github.com/sablier-labs/v2-core/blob/ea5110df0167a42595bdc7e3e3a6c649bc8f4e9c/src/abstracts/SablierV2Base.sol
